File: backend/app/schemas/people.py
from typing import Optional
from decimal import Decimal
from uuid import UUID
from pydantic import BaseModel


class PersonCreate(BaseModel):
    employee_name: str
    employee_id: Optional[str] = None
    wsp_role: Optional[str] = None
    team: Optional[str] = None
    role_on_project: Optional[str] = None
    hourly_rate: Optional[Decimal] = None
    years_experience: Optional[int] = None
    cv_path: Optional[str] = None


class PersonUpdate(BaseModel):
    employee_name: Optional[str] = None
    employee_id: Optional[str] = None
    wsp_role: Optional[str] = None
    team: Optional[str] = None
    role_on_project: Optional[str] = None
    hourly_rate: Optional[Decimal] = None
    years_experience: Optional[int] = None
    cv_path: Optional[str] = None


class PersonOut(BaseModel):
    id: UUID
    proposal_id: UUID
    employee_name: str
    employee_id: Optional[str]
    wsp_role: Optional[str]
    team: Optional[str]
    role_on_project: Optional[str]
    hourly_rate: Optional[Decimal]
    years_experience: Optional[int]
    cv_path: Optional[str]

    model_config = {"from_attributes": True}

File: frontend/src/api/people.ts
import api from "./client";

export interface Person {
  id: string;
  proposal_id: string;
  employee_name: string;
  employee_id: string | null;
  wsp_role: string | null;
  team: string | null;
  role_on_project: string | null;
  hourly_rate: number | null;
  years_experience: number | null;
  cv_path: string | null;
}

export const peopleApi = {
  list: (proposalId: string) =>
    api.get<Person[]>(`/api/proposals/${proposalId}/people/`).then(r => r.data),
  create: (proposalId: string, data: Partial<Person>) =>
    api.post<Person>(`/api/proposals/${proposalId}/people/`, data).then(r => r.data),
  update: (proposalId: string, personId: string, data: Partial<Person>) =>
    api.patch<Person>(`/api/proposals/${proposalId}/people/${personId}`, data).then(r => r.data),
  delete: (proposalId: string, personId: string) =>
    api.delete(`/api/proposals/${proposalId}/people/${personId}`),
};
